core/services/synchronization/explorer_client.go

```diff
@@ -130,7 +132,11 @@ func (ec *explorerClient) Send(data []byte) {
 	if !ec.started {
 		panic("send on unstarted explorer client")
 	}
-	ec.send <- data
+	select {
+	case ec.send <- data:
+	default:
+		logger.Warnw("explorer client buffer full, dropping message", "data", data)
```

Comment:
This could be super spammy. How about maintaining a count of how many messages were dropped and only logging a message if 
```
count > 0 && (count % 100 == 0 || count & (count - 1) == 0)
```
i.e. if the count is a multiple of 100 or a power of two. So we'd log at counts
```
1
2
4
8
16
32
64
100
200
300
...
```
The count would be reset to 0 when the first message makes it through again.

Reply:
Good idea!